pkg/storegateway/indexheader/index/postings.go

```diff
@@ -234,20 +231,16 @@ func readOffsetTable(factory *streamencoding.DecbufFactory, tableOffset int, f f
 	return d.Err()
 }
 
-func (t *PostingOffsetTableV1) PostingsOffset(name string, values ...string) ([]index.Range, error) {
-	rngs := make([]index.Range, 0, len(values))
+func (t *PostingOffsetTableV1) PostingsOffset(name string, value string) (index.Range, error) {
 	e, ok := t.postings[name]
 	if !ok {
-		return nil, nil
+		return index.Range{}, nil
 	}
-	for _, v := range values {
-		rng, ok := e[v]
-		if !ok {
-			continue
-		}
-		rngs = append(rngs, rng)
+	rng, ok := e[value]
+	if !ok {
+		return index.Range{}, nil
```

Comment:
How can the caller distinghuish whether the range exists or not? I see you're checking `rng.Start == 0 && rng.End == 0`, but maybe we could add a `bool` to the return values to signal whether exists or not (so it's more explicit). WDYT?

Reply:
Sounds good to me, I've added this in 55e71ab.